pkg/config/config.go

```diff
@@ -46,25 +46,25 @@ func (in *Install) DeepCopyInto(out *Install) {
 }
 
 type Registration struct {
-	URL             string            `json:"url,omitempty"`
-	CACert          string            `json:"ca_cert,omitempty"`
-	EmulateTPM      bool              `json:"emulateTPM,omitempty"`
-	EmulatedTPMSeed int64             `json:"emulatedTPMSeed,omitempty"`
-	NoSMBIOS        bool              `json:"noSMBIOS,omitempty"`
-	Labels          map[string]string `json:"labels,omitempty"`
+	URL             string            `json:"url,omitempty" yaml:"url,omitempty" mapstructure:"url"`
+	CACert          string            `json:"ca-cert,omitempty" yaml:"ca-cert,omitempty" mapstructure:"ca-cert"`
```

Comment:
understanding we needed a mapstructure tag here has been tricky! 😅

Reply:
OMG! At which stage do we use convert a struct to a map!? All this way more convoluted that it should, there should only be a single procedure to serialize and another one to deserialize 😅 

I am wondering why do we only need it for these two, it is weird that only two fields of the struct are dumped to or loaded from a map, while other not.

Reply:
the mapstructure package is used by viper to do the MergeInConfig [here](https://github.com/rancher/elemental-operator/blob/main/cmd/operator/register/root.go#L76), which processes the yaml file we put in the iso (which basically has only url and ca-cert values).
Only the url and the ca-cert are needed at that point to connect to the operator (cluster side). Once there we get the full yaml config, from which we do our "standard" yaml deserialization without passing through viper MergeInConfig 😅

Reply:
Oh yes, it's true, viper it's based on mapstructure, this why we have mapstructure and yaml labels in elemental-cli.